contract_manager/src/contracts/starknet.ts

```diff
@@ -65,29 +67,146 @@ export class StarknetPriceFeedContract extends PriceFeedContract {
     });
   }
 
-  getBaseUpdateFee(): Promise<{ amount: string; denom?: string | undefined }> {
-    throw new Error("Method not implemented.");
+  async getBaseUpdateFee(): Promise<{
+    amount: string;
+    denom?: string | undefined;
+  }> {
+    const tokens = await this.getFeeTokenAddresses();
+    return await this.getBaseUpdateFeeInToken(tokens[0]);
   }
-  getLastExecutedGovernanceSequence(): Promise<number> {
-    throw new Error("Method not implemented.");
+
+  /**
+   * Returns the list of accepted fee tokens.
+   * @returns hex encoded token addresses without 0x prefix
+   */
+  async getFeeTokenAddresses(): Promise<string[]> {
+    const contract = await this.getContractClient();
+    const tokens: bigint[] = await contract.fee_token_addresses();
+    return tokens.map((t) => t.toString(16));
   }
-  getPriceFeed(feedId: string): Promise<PriceFeed | undefined> {
-    throw new Error("Method not implemented.");
+
+  /**
+   * Returns the single update fee and symbol of the specified token.
+   * @param token hex encoded token address without 0x prefix
+   */
+  async getBaseUpdateFeeInToken(
+    token: string
+  ): Promise<{ amount: string; denom?: string | undefined }> {
+    token = "0x" + token;
+    const provider = this.chain.getProvider();
+    const contract = await this.getContractClient();
+    const fee: bigint = await contract.get_single_update_fee(token);
+
+    const tokenClassData = await provider.getClassAt(token);
+    const tokenContract = new Contract(tokenClassData.abi, token, provider);
+    const denom = shortString.decodeShortString(await tokenContract.symbol());
+    return { amount: fee.toString(), denom };
   }
-  executeUpdatePriceFeed(
+
+  async getLastExecutedGovernanceSequence(): Promise<number> {
+    const contract = await this.getContractClient();
+    return Number(await contract.last_executed_governance_sequence());
+  }
+
+  async getPriceFeed(feedId: string): Promise<PriceFeed | undefined> {
+    const contract = await this.getContractClient();
+    const result = await contract.query_price_feed_unsafe("0x" + feedId);
+    console.log(result);
+    if (result.Ok !== undefined) {
+      return {
+        price: convertPrice(result.Ok.price),
+        emaPrice: convertPrice(result.Ok.ema_price),
+      };
+    } else {
+      throw new Error(JSON.stringify(result.Err));
+    }
+  }
+
+  async executeUpdatePriceFeed(
     senderPrivateKey: PrivateKey,
     vaas: Buffer[]
   ): Promise<TxResult> {
-    throw new Error("Method not implemented.");
+    // We need the account address to send transactions.
+    throw new Error("Use executeUpdatePriceFeedWithAddress instead");
```

Comment:
Isn't there any way to fetch the address from private key?

Reply:
It's possible to derive it but it depends on the wallet application used to create the account and on the constructor arguments. Looks like the mainstream way of handling it is just to pass the account address explicitly.

Reply:
Since it's our own tooling, can't we just rely on that one?
if we had time, a better option would've been refactoring this code to have better interfaces.